index.js

```diff
@@ -39,16 +40,22 @@ function GeneratePath(type) {
       var parts = line.split(/\s+/g);
       if (parts.length > 7) {
         if (parts[11] && parts[11] !== '-') {
-          generatePath.push(cloudFrontDecode(parts[7] + '?' + parts[11]));
+          var path = cloudFrontDecode(parts[7] + '?' + parts[11]);
         } else {
-          generatePath.push(cloudFrontDecode(parts[7]));
+          path = cloudFrontDecode(parts[7]);
         }
+      } 
+      // get Referer
+      if (keepReferer && parts[9] && parts[9] !== '-') {
+        var referer = parts[9];
       }
+      if (path && referer) generatePath.push([path, referer]);
+      else if (path) generatePath.push(path);
```

Comment:
Nit: You might be able to reduce the if checks if you put lines 49-53 within the if block from lines 41-47.

Reply:
Good suggestion!